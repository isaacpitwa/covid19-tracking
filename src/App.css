@import url('https://fonts.googleapis.com/css2?family=Lato:wght@400;700&display=swap');
body{
  margin: 0;
  padding: 0;
  font-family: 'Lato', sans-serif;
}
header{
  display: flex;
  justify-content: space-between;
  padding: 8px 0%;
}
header h1, .select-dates h3{
 margin: 0;
}
.link {
  text-decoration: none;
}
ul {
  padding: 0;
  margin-top: 0;
}

.select-dates {
 display: flex;
 justify-content: center;
 /* align-items: center; */
 background-color: #DB4680;
 flex-direction: column;
 padding: 16px;
}
.select-dates div {
  display: flex;
  padding: 16px;
  padding-left: 0;
}
.select-dates div  input{
  width: 35%;
  height: 32px;
  border: 2px solid #fff;
  background-color: #DB4680;
  margin-right: 8px;
}
.select-dates div button {
  width: 30%;
}
.select-dates div button:hover {
  cursor: pointer;
  opacity: 0.6;
}
.home-heading {
  background-color: #CF4277;
  color: #fff;
  font-size: 16px;
  margin: 0;
  padding: 16px;
}

.metrics-list{
  margin: 0;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.metric {
  width: 25%;
  height: 32vh;
  background-color: #DB4680;
  color: #fff;
  border: 0;
  background-size: contain;
  background-repeat: no-repeat;
  position: relative;
}
.metric:nth-of-type(8n+2), .metric:nth-of-type(8n+4),.metric:nth-of-type(8n+5),.metric:nth-of-type(8n+7){
  background-color: #CF4277;
}
.metric:hover{
  cursor: pointer;
  opacity: 0.6;
}
.metric div {
  position: absolute;
  bottom: 28px;
  right: 28px;
  text-align: right;
}
.metric div  h4 {
  margin :0;
  font-size: 21px;
  font-family: 'Lato', sans-serif;
}
.metric div  p {
  margin :0;
  font-size: 14px;
  font-family: 'Lato', sans-serif;
}

/* Details */

.details-header {
  padding: 16px;
  display: flex;
  background-color: #FC5294;
  align-items: center;
  justify-content: center;
}
.details-header .img-bg{
  width: 70%;
  height: 20vh;
  background-size: contain;
  background-repeat: no-repeat;
}
.details-header .content {
  color: #fff;
 text-align: right;
 font-family: 'Lato', sans-serif;
}
.details-header .content h4 {
  margin: 0;
  font-size: 32px;
  font-weight: 700;
  font-family: 'Lato', sans-serif;
}

.details-header .content  p {
  margin :0;
  font-size: 18px;
  font-family: 'Lato', sans-serif;
}
.heading {
  background-color: #CF4277;
  color: #fff;
  font-size: 16px;
  padding: 21px 8px;
  text-transform: uppercase;
}

/* Region */
.region{
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: #fff;
  background-color: #DF4782;
  padding: 8px 21px;
  padding-right: 32px;
}
.region:nth-child(even){
  background-color:#CF4277;
}
.regions-list {
  margin: 0;
}
/* Utilities */

.page-heading {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #FC5294;
  color: #fff;
  padding: 0 16px;
}
.page-heading button {
  text-decoration: none;
  color: #fff;
  border: 0;
  background-color: #FC5294;

}
.page-heading button p{
 display: flex;
 align-items: center;
}
.page-heading h4 {
  margin: 0;
  font-weight: 400;
  font-size: 18px;
}
button:hover{
  cursor: pointer;
}