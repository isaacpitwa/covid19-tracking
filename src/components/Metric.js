import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { reset } from '../redux/countries/countries';

function Metric(props) {
  const { date, confirmedCases } = props;
  const navigate = useNavigate();
  const dispatch = useDispatch();

  return (
    <button
      className="link metric"
      onClick={() => {
        dispatch(reset());
        navigate(`/${date}/countries`);
      }}
      type="button"
      style={{ backgroundImage: 'url(images/map.png)' }}
    >
      <div>
        <h4>{date}</h4>
        <p>
          {confirmedCases.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')}
          {' '}
          Cases
        </p>
      </div>
    </button>
  );
}

Metric.propTypes = {
  date: PropTypes.string.isRequired,
  confirmedCases: PropTypes.number.isRequired,
};
export default Metric;
