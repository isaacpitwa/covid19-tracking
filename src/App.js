import { Route, Routes } from 'react-router-dom';
import './App.css';
import Countries from './components/Countries';
import CountryDetails from './components/CountryDetails';
import Metrics from './components/Metrics';

function App() {
  return (
    <Routes>
      <Route
        path="/"
        element={<Metrics />}
        exact
      />

      <Route
        path="/:date/countries"
        element={<Countries />}
        exact
      />
      <Route
        path="/:date/countries/:countryName"
        element={<CountryDetails />}
        exact
      />
    </Routes>
  );
}

export default App;
