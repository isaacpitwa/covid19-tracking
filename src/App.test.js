import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import store from './redux/configureStore';
import App from './App';
import Metric from './components/Metric';

const appRender = () => render(
  <Router>
    <Provider store={store}>
      <App />
    </Provider>
  </Router>,
);

describe('App', () => {
  it('renders without crashing', () => {
    appRender();
  });
  it('renders the Stats by Date', () => {
    appRender();
    expect(screen.getByText('Stats by Date')).toBeInTheDocument();
  });
  it('Renders Metric Componet', () => {
    const component = renderer.create(
      <Router>
        <Provider store={store}>
          <Metric key="2022-03-10" date="2022-03-10" confirmedCases="120" />
        </Provider>
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toBeDefined();
  });
});
