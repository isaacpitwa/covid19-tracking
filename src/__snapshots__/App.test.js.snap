// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Metric Componet 1`] = `
<button
  className="link metric"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(images/map.png)",
    }
  }
  type="button"
>
  <div>
    <h4>
      2022-03-10
    </h4>
    <p>
      120
       
      Cases
    </p>
  </div>
</button>
`;

exports[`Renders Country Renders Country Componet 1`] = `
<button
  className="link metric"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(images/map.png)",
    }
  }
  type="button"
>
  <div>
    <h4>
      2022-03-10
    </h4>
    <p>
      120
       
      Cases
    </p>
  </div>
</button>
`;

exports[`Renders Country Renders Metric Componet 1`] = `
<button
  className="link metric"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(images/map.png)",
    }
  }
  type="button"
>
  <div>
    <h4>
      2022-03-10
    </h4>
    <p>
      120
       
      Cases
    </p>
  </div>
</button>
`;

exports[`Renders Metric Renders Metric Componet 1`] = `
<button
  className="link metric"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(images/map.png)",
    }
  }
  type="button"
>
  <div>
    <h4>
      2022-03-10
    </h4>
    <p>
      120
       
      Cases
    </p>
  </div>
</button>
`;

exports[`Renders Metric Renders Metric Componet 2`] = `
<button
  className="link metric"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(images/map.png)",
    }
  }
  type="button"
>
  <div>
    <h4>
      2022-03-10
    </h4>
    <p>
      120
       
      Cases
    </p>
  </div>
</button>
`;
